Snowflake.Data/Core/RestParams.cs

```diff
@@ -60,13 +60,8 @@ static SFEnvironment()
             {
                 application = System.Diagnostics.Process.GetCurrentProcess().ProcessName,
                 osVersion = System.Environment.OSVersion.VersionString,
-#if NETFRAMEWORK
-                netRuntime = "NETFramework",

```

Comment:
I think you should extract this from runtime configuration.
See: https://learn.microsoft.com/en-us/dotnet/api/system.runtime.interopservices.runtimeinformation.frameworkdescription?view=net-6.0
The same applies to netVersion.
It would be good to extract it only once and then use it for every request.
It would be good if you wrote a test for proper netRuntime and netVersion extraction for given test environments.

Reply:
I was able to use `RuntimeInformation.FrameworkDescription` for the netRuntime property

However the version that comes from `RuntimeInformation.FrameworkDescription` seems different from what's expected:
- For .NET Framework it returns ".NET Framework 4.8.9246.0" for all the versions instead of ".NET Framework 471", ".NET Framework 472", or ".NET Framework 481"
- For .NET 6.0 and greater it returns ".NET 8.0.6" instead of ".NET 6.0", ".NET 7.0", or ".NET 8.0"

Is that the expected behavior?

Reply:
not really... maybe try to get the version somehow differently from the assembly.

Reply:
I tried to use the version from the assembly but it returned "netstandard2.0" since that's the target framework of the project . Should the `netVersion` value be set to `2.0` or `netstandard2.0` since that is the target version?

Reply:
It's fixed now for .NET 6.0 and above

I'm not sure why the .NET Framework versions still returns "4.8" but it might be because it installs the latest .NET Framework version that is [compatible with previous versions](https://support.microsoft.com/en-us/topic/microsoft-net-framework-4-8-offline-installer-for-windows-9d23f658-3b97-68ab-d013-aa3c3e7495e0)
> The Microsoft .NET Framework 4.8 is a highly compatible, in-place update to the Microsoft .NET Framework 4, 4.5, 4.5.1, 4.5.2, 4.6, 4.6.1, 4.6.2, 4.7, 4.7.1 and 4.7.2.

Reply:
The `Assembly.GetExecutingAssembly().Location` contains the accurate .NET Framework version and is now used for the `netVersion` value

Just for future reference, here's a list of what was previously tried and their results when running on .NET Framework:
1. `TargetFrameworkAttribute` of the executing assembly returns "netstandard2.0"
2. `TargetFrameworkAttribute` of the calling assembly returns "netstandard2.0"
3. The entry assembly returns null
4. `System.Environment` property returns "4.0.30319.42000"
5. `RuntimeInformation.FrameworkDescription` returns "4.8.4724.0"

Reply:
It's fixed for both .NET 6.0+ and .NET Framework versions now

Reply:
as I see you extract the location of the assembly and get the information from the directory name.
I don't like this idea since the library can be located anywhere - we don't have a guarantee that customers would store their dlls in similar form as it is when building the project.

Try maybe something else. If you would not find anything better I think using RuntimeInformation.FrameworkDescription would be better than using location of the dll.

Reply:
It's reverted to use the `RuntimeInformation.FrameworkDescription` again. I was not able to find a way to get framework version from other methods